# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveUtil.h
# Commit: b1fda3f9af8a
# Full Hash: b1fda3f9af8ae58da0ea4a24078dd75b784e79cf
# Author: Cosmin Sabou <csabou@mozilla.com>
# Date: 2025-09-10 09:37:11
# Regressor Bug: 1979588
# File Overlap Count: 2
# Overlapping Files: dom/base/test/test_text-fragments-create-text-directive.html, dom/base/TextDirectiveCreator.cpp
# Description:
#   Revert "Bug 1979588 - Text Fragments: Correctly deal with long words at end of range for range-based matching. r=dom-core,smaug" for causing crashes (bug 1987845) as part of the Nightly merge operations
#   
#   This reverts commit 7db1bf44de9eb0bcf7b68ff1b5206fc801432ee0.
# ==============================================================================

diff -r 89d6d9a06132 -r b1fda3f9af8a dom/base/TextDirectiveUtil.h
--- a/dom/base/TextDirectiveUtil.h	Wed Sep 10 01:47:49 2025 +0000
+++ b/dom/base/TextDirectiveUtil.h	Wed Sep 10 12:33:12 2025 +0300
@@ -551,10 +551,8 @@
     return 0;
   }
 
-  MOZ_DIAGNOSTIC_ASSERT(
-      !nsContentUtils::IsHTMLWhitespace(aReferenceString.First()));
-  MOZ_DIAGNOSTIC_ASSERT(
-      !nsContentUtils::IsHTMLWhitespace(aReferenceString.Last()));
+  MOZ_ASSERT(!nsContentUtils::IsHTMLWhitespace(aReferenceString.First()));
+  MOZ_ASSERT(!nsContentUtils::IsHTMLWhitespace(aReferenceString.Last()));
   uint32_t referenceStringPosition =
       direction == TextScanDirection::Left ? aReferenceString.Length() - 1 : 0;
 